protocols/mdns/src/behaviour/iface.rs

```diff
@@ -217,6 +289,12 @@ where
 
                     self.discovered
                         .extend(response.extract_discovered(Instant::now(), self.local_peer_id));
+
+                    // Stop probing when we have a valid response
+                    if !self.discovered.is_empty() {
+                        self.probe_state.update(true, &self.query_interval);
```

Comment:
Wouldn't it be easier to understand to assign `probe_state` directly to a new variant here?

Reply:
I agree. Code is much simpler now